test/cufinufft2d1nupts_test.cu

```diff
@@ -252,5 +252,9 @@ int main(int argc, char* argv[])
 	cudaFree(d_y2);
 	cudaFree(d_c2);
 	cudaFree(d_fk2);
+	
+	// for cuda-memcheck to work
+	cudaDeviceReset();
```

Comment:
Is this something we should be doing in the other tests as well?

Reply:
I don't think it is necessary. According to this stack overflow post: https://stackoverflow.com/questions/36012289/what-is-the-role-of-cudadevicereset-in-cuda, the call of the function is not considered mandatory. But as the discussions in the answer suggest, it is required for using tools like `cuda-memcheck` or using the profilers.

Reply:
Ok sounds good.